docs/kubernetes/ingress-backends.md

```diff
@@ -0,0 +1,70 @@
+# Kubernetes Backend Deployments
+
+## Kubernetes Race Condition problem
+
+As described in [#652](https://github.com/zalando/skipper/issues/652),
+there is a problem that exists in kubernetes, while terminating Pods.
+Terminating Pods could be graceful, but the nature of distributed
+environments will show failures, because not all components in the
+distributed system changed already their state. When a Pod terminates,
+the controller-manager has to update the `endpoints` of the kubernetes
+`service`.  Additionally skipper has to get this endpoints
+list. Skipper polls the kube-apiserver every `-source-poll-timeout=<ms>`,
+which defaults to 3000.
+Reducing this interval or implementing watch will only reduce the
+timeframe, but not fix the underlying race condition.
+
+Mitigation strategies can be different and the next section document
+strategies for application developers to mitigate the problem.
+
+## Teardown strategies
+
+An application that is target of an ingress can circumvent HTTP code
+504s Gateway Timeouts with these strategies:
+
+1. use [Pod lifecycle hooks](#pod-lifecycle-hooks)
+2. use a SIGTERM handler to switch `readinessProbe` to unhealthy and
+teardown your application later, probably just waiting until SIGKILL
```

Comment:
The lead sentence starts like this: "An application that is target of...". So instead of "...and teardown your application, ...", we could write "...and exit later, or just wait for SIGKILL terminating the process."